src/Shared/FileMatcher.cs

```diff
@@ -1673,7 +1673,7 @@ internal static bool IsMatch(string input, string pattern)
         /// <param name="pattern">Pattern against which string is matched.</param>
         internal static bool IsMatch(ReadOnlySpan<char> input, string pattern)
         {
-            if (input == null)
+            if (input == ReadOnlySpan<char>.Empty) // Comparing ReadOnlySpan with null in this scenario transforms the null to ReadOnlySpan Empty (https://github.com/dotnet/runtime/issues/84265)
```

Comment:
I am glad to have this explanation and link now but I'm not sure it deserves a comment.

```suggestion
            if (input == ReadOnlySpan<char>.Empty)
```